subsys/net/ip/net_core.c

```diff
@@ -29,6 +29,7 @@ LOG_MODULE_REGISTER(net_core, CONFIG_NET_CORE_LOG_LEVEL);
 #include <net/tcp.h>
 #include <net/gptp.h>
 #include <net/lldp.h>
+#include <net/ethernet.h>
```

Comment:
Ethernet in net_core? Oh, no! @tbursztyka did so good work to separate L2's from generic networking and it creeps back now?

Reply:
Only to get ETH_P_ALL which is used in raw sockets to inform that the caller wants to receive all packets.

Reply:
> Only to get ETH_P_ALL which is used in raw sockets to inform that the caller wants to receive all packets.

Not sure if it's needed. That header can be included in sockets_packet.c, but I don't see a good reason why it should be included in net/ip/net_core.c.

Reply:
Provided separate file raw.c and raw.h